axiom/nr_distributed_trace.c

```diff
@@ -481,14 +481,28 @@ void nr_distributed_trace_set_app_id(nr_distributed_trace_t* dt,
 }
 
 void nr_distributed_trace_set_trace_id(nr_distributed_trace_t* dt,
-                                       const char* trace_id) {
+                                       const char* trace_id,
+                                       bool do_padding) {
   if (NULL == dt) {
     return;
   }
 
   nr_free(dt->trace_id);
   if (trace_id) {
-    dt->trace_id = nr_strdup(trace_id);
+    if (do_padding) {
+      int len = nr_strlen(trace_id);
+      if (len < NR_TRACE_ID_SIZE) {
+        int padding = NR_TRACE_ID_SIZE - len;
+        char* dest = (char*)nr_malloc(NR_TRACE_ID_SIZE + 1);
+        for (int i=0; i < padding; i++) {
+          dest[i] = '0';
+        }
+        nr_strcpy(dest + padding, trace_id);
```

Comment:
`nr_strcpy` uses `strlen` (see [here](https://github.com/newrelic/newrelic-php-agent/blob/R10.22/axiom/util_strings.c#L154)) and therefore is less efficient than just `strcpy`. Please use `strcpy`.

Reply:
changed in [8c5cf32](https://github.com/newrelic/newrelic-php-agent/pull/929/commits/8c5cf329b62381992220378ad576d58752ed3b5e)